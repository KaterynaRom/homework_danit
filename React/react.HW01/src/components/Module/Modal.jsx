import React, {PureComponent} from 'react';
import styles from './Modal.module.scss';

class Modal extends React.PureComponent {
  render(){
    const { title, text, closeModal, actions, closeButton } = this.props;
    console.log(this.props)

    return (
      <div className={styles.background} onClick={closeModal}>
        <div className={styles.modal}>
          {closeButton}
          <h3 className={styles.title}>{title}</h3>
          <div className={styles.text}>{text}</div>
          <div className={styles.buttons}>{actions}</div>
        </div>
      </div>
    );
  }
};

Modal.propTypes = {};

Modal.defaultProps = {};

export default Modal;
