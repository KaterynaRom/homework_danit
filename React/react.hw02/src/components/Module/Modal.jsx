import React, {PureComponent} from 'react';
import styles from './Modal.module.scss';

class Modal extends React.PureComponent {

  render () {
    const { header, text, closeModal, actions, closeButton } = this.props;

    return (
      <div className={styles.modalWrapper}>
        <div className={styles.background} onClick={closeModal}/>
        <div className={styles.modal}>
          {closeButton}
          <div className={styles.titleWrapper}>
            <div className={styles.title}>{header}</div>
          </div>
          <div className={styles.text}>{text}</div>
          <div className={styles.buttons}>{actions}</div>
        </div>
      </div>

    );
  }
}

Modal.propTypes = {};

Modal.defaultProps = {};

export default Modal;
