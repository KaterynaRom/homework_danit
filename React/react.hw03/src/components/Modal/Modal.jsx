import React from 'react';
import styles from './Modal.module.scss';
import PropTypes from 'prop-types';

const Modal = (props) => {
  const { header, text, closeModal, actions, closeButton } = props;

  return (
    <div className={styles.modalWrapper}>
      <div className={styles.background} onClick={closeModal}/>
      <div className={styles.modal}>
        {closeButton}
        <div className={styles.titleWrapper}>
          <div className={styles.title}>{header}</div>
        </div>
        <div className={styles.text}>{text}</div>
        <div className={styles.buttons}>{actions}</div>
      </div>
    </div>

  );
}

Modal.propTypes = {
  header: PropTypes.string,
  text: PropTypes.string,
  closeModal: PropTypes.func,
  closeButton: PropTypes.func,
}

Modal.defaultProps = {
  header: 'information',
  text: 'Confirm your actions.',
  closeModal: () => {},
  closeButton: () => {},
}

export default Modal;
