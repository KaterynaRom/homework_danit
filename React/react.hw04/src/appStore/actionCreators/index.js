import {
      GET_PRODUCTS_DATA,
      SAVE_CART_FROM_LS,
      TOGGLE_IS_FAVOURITE_CARDS,
      ADD_TO_CART,
      REMOVE_FROM_CART,
      SET_IS_OPEN_MODAL,
      SET_MODAL_CONTENT} from "../actions";
import {cartStatusLS} from "../../utils/utils";



export const getProductsData = () => async (dispatch) => {
      const response = await fetch('./productsList.json')
        .then(response => response.json());
      const favStatus = localStorage.getItem('favourites') || [];
      response.forEach(prod => prod.isFavourite = favStatus.includes(prod.code));
            dispatch(saveCartFromLS(cartStatusLS));
      dispatch({type: GET_PRODUCTS_DATA, payload: response})
}

const saveCartFromLS = (cartFromLS) => ({type: SAVE_CART_FROM_LS, payload: cartFromLS});

export const toggleIsFavourite = (products) => (dispatch) => {
            dispatch(getProductsData());
      dispatch({type: TOGGLE_IS_FAVOURITE_CARDS, payload: products});
}

export const addToCart = (cartProduct) => ({
      type: ADD_TO_CART, payload: cartProduct
})

export const setIsOpenModal = (isOpenModal) => ({type: SET_IS_OPEN_MODAL, payload: isOpenModal});
export const setModalContent = (isAddModal) => ({type: SET_MODAL_CONTENT, payload: isAddModal});


