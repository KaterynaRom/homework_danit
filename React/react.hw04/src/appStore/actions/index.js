//products
export const GET_PRODUCTS_DATA = 'GET_PRODUCTS_DATA';

//favourite
export const TOGGLE_IS_FAVOURITE_CARDS = 'TOGGLE_IS_FAVOURITE_CARDS';

//cart
export const ADD_TO_CART = 'ADD_TO_CART';
export const REMOVE_FROM_CART = 'REMOVE_FROM_CART';
export const SAVE_CART_FROM_LS = 'SAVE_CART_FROM_LS';


export const SET_IS_OPEN_MODAL = 'SET_IS_OPEN_MODAL';
export const SET_MODAL_CONTENT = 'SET_MODAL_CONTENT';