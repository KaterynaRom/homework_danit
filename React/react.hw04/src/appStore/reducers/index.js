import {combineReducers} from 'redux';
import cartReducer from './cartReducer';
import cardReducer from './cardReducer';
import modalReducer from './modalReducer';

const reducer = combineReducers({
  card: cardReducer,
  cart: cartReducer,
  modal: modalReducer,
})
export default reducer