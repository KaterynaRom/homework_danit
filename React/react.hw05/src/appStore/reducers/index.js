import {combineReducers} from 'redux';
import cartReducer from './cartReducer';
import cardReducer from './cardReducer';
import modalReducer from './modalReducer';
import userReducer from "./userReducer";

const reducer = combineReducers({
  card: cardReducer,
  cart: cartReducer,
  modal: modalReducer,
  user: userReducer,
})
export default reducer