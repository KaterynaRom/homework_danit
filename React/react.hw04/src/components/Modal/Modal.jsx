import React from 'react';
import styles from './Modal.module.scss';
import {setIsOpenModal, removeFromCart} from "../../appStore/actionCreators";
import {useDispatch, useSelector} from "react-redux";


const Modal = () => {
  const {isOpenModal, modalTitle, code } = useSelector(state => state.modal);
  const dispatch = useDispatch();

  const closeModal = () => {dispatch(setIsOpenModal(false))}

  if (!isOpenModal) return null;

  return (
    <div className={styles.modalWrapper}>
      <div className={styles.background} onClick={closeModal}/>

      <div className={styles.modal}>
        <button onClick={closeModal}>âœ–</button>
        <div className={styles.titleWrapper}>
          <div className={styles.title}>Confirm Action</div>
        </div>
        <div className={styles.text}>{modalTitle}</div>
        <div className={styles.buttons}>
          <button
            onClick={()=> {
              dispatch(removeFromCart(code));
              closeModal();
            }}
            >Yes</button>

          <button onClick={closeModal}
          >No</button>
        </div>
      </div>

    </div>
  );
}

export default Modal;
