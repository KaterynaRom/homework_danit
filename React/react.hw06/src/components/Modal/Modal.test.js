import Modal from './Modal';
import { render } from '@testing-library/react';
import store from "../../appStore";
import {Provider, useDispatch} from "react-redux";
import {setModalParams, setIsOpenModal} from "../../appStore/actionCreators";

const Component = () => {
  const dispatch = useDispatch();

  dispatch(setModalParams('testCode'));
  dispatch(setIsOpenModal(true));

  return (<Modal />)
}

describe("Modal render", () => {
  test("should modal render", ()=> {
    const {asFragment} = render(
      <Provider store={store}>
        <Component/>
      </Provider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
});

