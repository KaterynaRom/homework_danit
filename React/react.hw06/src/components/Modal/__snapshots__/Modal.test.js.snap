// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal render should modal render 1`] = `
<DocumentFragment>
  <div
    class="modalWrapper"
  >
    <div
      class="background"
    />
    <div
      class="modal"
    >
      <button>
        âœ–
      </button>
      <div
        class="titleWrapper"
      >
        <div
          class="title"
        >
          Confirm Action
        </div>
      </div>
      <div
        class="text"
      >
        Are you sure you want to remove the product?
      </div>
      <div
        class="buttons"
      >
        <button>
          Yes
        </button>
        <button>
          No
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
